ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh

```diff
@@ -0,0 +1,408 @@
+    echo "Command: $cmd"
+    status=$(eval "$cmd")
+    echo "Status: $status"
+    sleep $interval
+    try=$(expr $try + 1)
+done
+if [ X"$status" != X"success" ]; then
+    echo  "Something wrong, copy timeout or failed!"
+    exit 2
+fi
+vhd_blob_url=$(az storage blob url --account-name ${sa_name} --account-key ${account_key} -c ${storage_contnainer} -n ${vhd_name} -o tsv)
+
+echo "Deploy the bastion image from bastion vhd"
+run_command "az image create --resource-group ${bastion_rg} --name '${bastion_name}-image' --source ${vhd_blob_url} --os-type Linux --storage-sku Standard_LRS" || exit 2
+bastion_image_id=$(az image show --resource-group ${bastion_rg} --name "${bastion_name}-image" | jq -r '.id')
+
+echo "Create bastion subnet"
+open_port="22 3128 3129 5000" bastion_nsg="${bastion_name}-nsg" bastion_subnet="${bastion_name}Subnet"
+run_command "az network nsg create -g ${bastion_rg} -n ${bastion_nsg}" &&
+run_command "az network nsg rule create -g ${bastion_rg} --nsg-name '${bastion_nsg}' -n '${bastion_name}-allow' --priority 1000 --access Allow --source-port-ranges '*' --destination-port-ranges ${open_port}" &&
+#subnet cidr for int service is hard code, it should be a sub rang of the whole VNet cidr, and not conflicts with master subnet and worker subnet
+bastion_subnet_cidr="10.0.99.0/24"
+vnet_subnet_address_parameter="--address-prefixes ${bastion_subnet_cidr}"
+run_command "az network vnet subnet create -g ${bastion_rg} --vnet-name ${bastion_vnet_name} -n ${bastion_subnet} ${vnet_subnet_address_parameter} --network-security-group ${bastion_nsg}" || exit 2
+
+echo "Create bastion vm"
+bastion_ignition_file="${vhd_data}"
+run_command "az vm create --resource-group ${bastion_rg} --name ${bastion_name} --admin-username core --admin-password 'NotActuallyApplied!' --image '${bastion_image_id}' --os-disk-size-gb 99 --subnet ${bastion_subnet} --vnet-name ${bastion_vnet_name} --nsg '' --size 'Standard_DS1_v2' --debug --custom-data '${bastion_ignition_file}'" || exit 2
+
+#####################################
+#########Save Bastion Info###########
+#####################################
+bastion_internal_ip=$(az vm list-ip-addresses --name ${bastion_name} --resource-group ${bastion_rg} | jq -r ".[].virtualMachine.network.privateIpAddresses[]") &&
+bastion_public_ip=$(az vm list-ip-addresses --name ${bastion_name} --resource-group ${bastion_rg} | jq -r ".[].virtualMachine.network.publicIpAddresses[].ipAddress") || exit 2
+if [ X"$bastion_public_ip" == X"" ] || [ X"$bastion_internal_ip" == X"" ] ; then
+    echo "Did not found public or internal IP!"
+    exit 1
+fi
+
+proxy_credential=$(cat "${src_proxy_creds_file}")
+PUBLIC_PROXY_URL="http://${proxy_credential}@${bastion_public_ip}:3128"
+INTERNAL_PROXY_URL="http://${proxy_credential}@${bastion_internal_ip}:3128"
+
+
+echo "${INTERNAL_PROXY_URL}" > "${SHARED_DIR}/internal_proxy_url"
+
+cat > "${SHARED_DIR}/client_proxy_setting.sh" << EOF
```

Comment:
Agree with a consistent file name. After reviewed the linked PR, seem like once `proxy-conf.sh` exists,the file will always be sourced, then the cluster will be always treated as a private cluster, while this command.sh is some kinds of pre actions, can not determinate if it is a private cluster, so that means `proxy-conf.sh` should be created in some ipi-conf-azure workflow once confirm it is a private cluster.

Reply:
A follow up PR like https://github.com/openshift/release/pull/26672

Reply:
or, is that possible that create an individual step for creating `proxy-conf.sh`? It allows users control whether create it or not.

Reply:
It sounds good to me to introduce an individual step for creating proxy-conf.sh. Let us leave that to private cluster workflow implement. Or maybe you can implement it in your govus workflow, later other one can reference the step in their workflow implementation.

Reply:
sure, I will create a separate step.